src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/Operations/CosmosFhirOperationDataStore.cs

```diff
@@ -169,9 +171,9 @@ public async Task<ExportJobOutcome> GetExportJobByHashAsync(string hash, Cancell
             }
             catch (CosmosException dce)
             {
-                if (dce.StatusCode == HttpStatusCode.TooManyRequests)
+                if (dce.InnerException is RequestRateExceededException)
```

Comment:
Perhaps we can smooth this over a little with some extensions...
`bool dce.IsRateExceededException()`?
Or similar, I guess the point might be to unwrap inside an extension, keeps the logic consistent everywhere

Reply:
There is something about accepting this to be the inner exception that just doesn't feel right. And I don't know that adding an extension method helps that much (why should write `catch (Exception e) when (e.IsRequestRateExceededException())` instead of `catch (RequestRateExceededException)`?). I feel like this fix is a stop-gap, but I don't find it very satisfying.

Reply:
Should another data provider throw RequestRateExceededException as an inner exception too? Seems like the answer should be no.

Reply:
...But I've added the extension methods :)

Reply:
Agree, it seems like the simplest fix, yet unsatisfying. Though if it improves the experience we have time to find a nicer solution